testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/SecurityAuthCommandsTestCase.java

```diff
@@ -0,0 +1,876 @@
+
+        Assert.assertEquals(TEST_FS_REALM, getMechanismRealmMapper(true, "DIGEST-MD5"));
+        getNames(Util.CONSTANT_REALM_MAPPER).contains(TEST_FS_REALM);
+
+        // check that domain has both realms.
+        List<String> realms = getDomainRealms(TEST_DOMAIN);
+        Assert.assertTrue(realms.contains("ManagementRealm"));
+        Assert.assertTrue(realms.contains(TEST_FS_REALM));
+    }
+
+    @Test
+    public void testOOBHTTPManagement() throws Exception {
+        ctx.handle("security enable-http-auth-management --no-reload");
+        Assert.assertEquals(ElytronUtil.OOTB_MANAGEMENT_HTTP_FACTORY, getManagementInterfaceAuthFactory(ctx, null, false));
+        ctx.handle("security disable-http-auth-management --no-reload");
+        Assert.assertNull(getManagementInterfaceAuthFactory(ctx, null, true));
+    }
+
+    @Test
+    public void testOOBHTTPManagement2() throws Exception {
+        // New factory but reuse OOB ManagementRealm properties realm.
+        // side effect is to create a constant realm mapper for ManagementRealm.
+        ctx.handle("security enable-http-auth-management --no-reload --mechanism=BASIC "
+                + "--user-properties-file=mgmt-users.properties --group-properties-file=mgmt-groups.properties  "
+                + "--relative-to=jboss.server.config.dir --exposed-realm=ManagementRealm --new-security-domain-name=" + TEST_DOMAIN
+                + " --new-auth-factory-name=" + TEST_HTTP_FACTORY);
+        Assert.assertEquals(TEST_HTTP_FACTORY, getManagementInterfaceAuthFactory(ctx, null, false));
+
+        // Check Realm.
+        Assert.assertEquals("ManagementRealm", getExposedRealmName(false, "BASIC"));
+        Assert.assertEquals("ManagementRealm", getMechanismRealmMapper(false, "BASIC"));
+        getNames(Util.CONSTANT_REALM_MAPPER).contains("ManagementRealm");
+
+        // Replace with file system realm.
+        ctx.handle("security enable-http-auth-management --no-reload --mechanism=BASIC "
+                + "--file-system-realm-name=" + TEST_FS_REALM);
+
+        Assert.assertEquals(TEST_FS_REALM, getMechanismRealmMapper(false, "BASIC"));
+        getNames(Util.CONSTANT_REALM_MAPPER).contains(TEST_FS_REALM);
+
+        // check that domain has both realms.
+        List<String> realms = getDomainRealms(TEST_DOMAIN);
+        Assert.assertTrue(realms.contains("ManagementRealm"));
+        Assert.assertTrue(realms.contains(TEST_FS_REALM));
+    }
+
+    @Test
+    public void testSASLManagement() throws Exception {
+        String cmd = "security enable-sasl-management --no-reload --mechanism=DIGEST-MD5"
+                + " --user-properties-file=mgmt-users.properties --group-properties-file=mgmt-groups.properties"
+                + " --relative-to=jboss.server.config.dir --new-security-domain-name=" + TEST_DOMAIN
+                + " --new-auth-factory-name=" + TEST_SASL_FACTORY + " --new-realm-name=" + TEST_USERS_REALM;
+        {
+            // Must use --exposed-realm-name=
```

Comment:
Why is exposed realm required?

Reply:
It is required because it must match exactly the one present inside the properties file. I am thinking that, in order to avoid complex debugging (this detail is tricky...) the user should provide explicitly the exposed realm when using properties realm.